### PR TITLE
Follow the conventional model_name

### PR DESCRIPTION
**Pull request by AllentDan**
#1674

### CODE REVIEW

File: lmdeploy/serve/async_engine.py [hunk 1]
```diff
@@ -55,12 +54,10 @@ def _config_model_name(config):
         model_name = get_model_name_from_workspace_model(model_path)
     # may get a model name from model_path
     if model_name is None:
-        model_name = best_match_model(model_path)
-        if model_name is None:
-            raise ArgumentError(None,
-                                f'Please set model_name for {model_path}')
+        if os.path.exists(model_path):
```
Comment (lvhan028) [inline lmdeploy/serve/async_engine.py:57]: https://github.com/vllm-project/vllm/blob/e07aff9e52342dc82b73c803ba69601242801bc4/vllm/entrypoints/openai/api_server.py#L180-L181

Comment (AllentDan) [reply lmdeploy/serve/async_engine.py:57]: The condition line is above.

**Comment by lvhan028** [comment]
···shell
lmdeploy serve api_server /workspace/models-140/InternLM/internlm2-chat-7b
```
but got model_name "internvl-internlm2"

File: lmdeploy/serve/async_engine.py (hunk 1 above)
Comment (lvhan028) [reply lmdeploy/serve/async_engine.py:57]: I mean in vLLM the served_model_name is the model_path if it is not specified

**Comment by lvhan028** [comment]
@grimoire test_mbgmv failed, could you help check it?
https://github.com/InternLM/lmdeploy/actions/runs/9298553214/job/25590670675?pr=1677#step:10:91

**Review by lvhan028 — APPROVED** [review]

**Comment by grimoire** [comment]
The failure comes from low threshold. It is safe to just ignore the error or enlarge `atol`.

**Review by irexyc — APPROVED** [review]

**Pull request merged**